services/context/repo.go

```diff
@@ -988,6 +1004,8 @@ func RepoRefByType(detectRefType RepoRefType, opts ...RepoRefByTypeOptions) func
 
 		ctx.Data["BranchName"] = ctx.Repo.BranchName
 		ctx.Data["RefName"] = ctx.Repo.RefName
+		ctx.Data["RefShortName"] = util.Iif(refType == RepoRefCommit, base.ShortSha(ctx.Repo.RefName), ctx.Repo.RefName)
+		ctx.Data["RefType"] = RefTypeName(refType)
```

Comment:
We shouldn't do that (keep adding more variables to templates)

If you'd like to refactor it, it needs a separate PR, mixing everything together would make this PR unmaintainable and possible to lose code.